controllers/storageclient_controller.go

```diff
@@ -203,9 +205,54 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	if err := r.reconcileClusterResourceQuota(externalClusterClient); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileClusterResourceQuota(externalClusterClient *providerClient.OCSProviderClient) error {
```

Comment:
not sure this is enough, there could be more than 1 storageclient and you can't scope a single clusterquota resource to entire cluster but scope it to sum of all storageclasses that reference this storageclient and does this quota also takes snapshots into account?

Reply:
it won't consider volume snapshot because Snapshot YAML has the source field(inside spec section) pointing to PVC.

```
spec:
  source:
    persistentVolumeClaimName: pvc-test
```
and PVC spec has pointing `requests.storage`
```
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 20Gi
```
In CRC we are mentioning `requests.storage` so it will consider only PVC which has `request.storage` in it.

Reply:
> not sure this is enough, there could be more than 1 storageclient and you can't scope a single clusterquota resource to entire cluster but scope it to sum of all storageclasses that reference this storageclient

Regarding this this is the limitation using ClusterResourceQuota and Eran is aware of it. @nb-ohad please correct me if I am wrong.

Reply:
iirc, the limitation that's discussed was more than one storageclaim referencing a single storageclient but not more than one storageclient is delimited by a single clusterresourcequota.

imho, from the start we are taking a holistic approach for accommodating more than one storageclient and deviating from that might create problems if we use multiple storageclients.